### ISSUE TITLE
Failed to watch xxx: failed to list xxx net/http: TLS handshake timeout

### ISSUE THREAD

**Issue by Poorunga** [kubeedge/edgemesh#59]
**What happened**:

When I deployed `edgemesh-agent-cloud`, I found that it did not work properly. And when I execute `docker logs -f <edgemesh_container_id>`:
```
I0824 12:14:27.885528       1 trace.go:205] Trace[1396859110]: "Reflector ListAndWatch" name:/code/common/informers/informers.go:87 (24-Aug-2021 12:14:17.883) (total time: 10001ms):
Trace[1396859110]: [10.001600982s] [10.001600982s] END
E0824 12:14:27.885553       1 reflector.go:127] /code/common/informers/informers.go:87: Failed to watch *v1.Service: failed to list *v1.Service: Get "https://10.96.0.1:443/api/v1/services?limit=500&resourceVersion=0": net/http: TLS handshake timeout
I0824 12:14:27.885771       1 trace.go:205] Trace[1046529680]: "Reflector ListAndWatch" name:/code/common/informers/informers.go:87 (24-Aug-2021 12:14:17.883) (total time: 10002ms):
Trace[1046529680]: [10.002072265s] [10.002072265s] END
E0824 12:14:27.885780       1 reflector.go:127] /code/common/informers/informers.go:87: Failed to watch *v1alpha3.DestinationRule: failed to list *v1alpha3.DestinationRule: Get "https://10.96.0.1:443/apis/networking.istio.io/v1alpha3/destinationrules?limit=500&resourceVersion=0": net/http: TLS handshake timeout
```

The log of edgemesh-agent-cloud will print a lot of `net/http: TLS handshake timeout`.

**Why?**

This may be due to the residual iptables or  ip route rules.

1. Iptables rules remain like:
```
// in edgemesh-agent-cloud container
/ # iptables -t nat -nvL
Chain PREROUTING (policy ACCEPT 83 packets, 3968 bytes)
 pkts bytes target     prot opt in     out     source               destination
   83  3968 EDGE-MESH  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* edgemesh root chain */
    0     0 EDGE-MESH  tcp  --  *      *       0.0.0.0/0            10.96.0.0/12

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 5 packets, 342 bytes)
 pkts bytes target     prot opt in     out     source               destination
   50  3042 EDGE-MESH  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* edgemesh root chain */
   81  4860 EDGE-MESH  tcp  --  *      *       0.0.0.0/0            10.96.0.0/12

Chain POSTROUTING (policy ACCEPT 50 packets, 3042 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain EDGE-MESH (4 references)
 pkts bytes target     prot opt in     out     source               destination
   45  2700 EDGE-MESH-TCP  tcp  --  *      *       0.0.0.0/0            10.96.0.0/12         /* tcp service proxy */

Chain EDGE-MESH-TCP (1 references)
 pkts bytes target     prot opt in     out     source               destination
   45  2700 DNAT       tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            to:10.244.1.1:40001
```

PREROUTING chain and OUTPUT chain will have two EDGE-MESH targets. Then we need to delete the second rule later, for example:
**Note: 2 is my iptables rule index, which may not be on your host!!!**

```
iptables -t NAT -D PREROUTING 2
iptables -t NAT -D OUTPUT 2
```

2. IP ruote table rules remain like:
```
[root@k8s-node1 ~]# ip route show
default via 192.168.0.1 dev eth0 proto dhcp metric 100
10.96.0.0/12 via 10.244.1.1 dev cni0
10.244.0.0/24 via 10.244.0.0 dev flannel.1 onlink
10.244.1.0/24 dev cni0 proto kernel scope link src 10.244.1.1
169.254.169.254 via 192.168.0.254 dev eth0 proto dhcp metric 100
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown
192.168.0.0/24 dev eth0 proto kernel scope link src 192.168.0.71 metric 100
```

The second route rule `10.96.0.0/12 via 10.244.0.1 dev cni0` is residual. Generally speaking, route rule will be cleared automatically when edgemesh-agent is shut down. If not, you need to clear it manually, for example:
```
ip route del 10.96.0.0/12 via 10.244.1.1 dev cni0
```

Finally, redeploy edgemesh-agent.

**Comment by Poorunga** [comment]
The latest edgemesh-agent and can guarantee that these things will not happen. 👏

**Comment by Poorunga** [comment]
fix PR #63 

/close
